### PR TITLE
ExtSavedata: Save the icon passed to CreateExtSaveData.

### PR DESCRIPTION
**Pull request by Subv**
Organize the ExtSaveData folders as they are stored in the console.

### CODE REVIEW

File: src/core/hle/service/fs/archive.cpp [hunk 1]
```diff
@@ -421,9 +421,25 @@ ResultCode CreateExtSaveData(MediaType media_type, u32 high, u32 low) {
...
+        return ResultCode(-1); // TODO(Subv): Find the right error code
+    if (!FileUtil::CreateFullPath(boss_path))
+        return ResultCode(-1); // TODO(Subv): Find the right error code
+
+    u8* smdh_icon = Memory::GetPointer(icon_buffer);
+    if (!smdh_icon)
         return ResultCode(-1); // TODO(Subv): Find the right error code
+
+    // Create the icon
+    FileUtil::IOFile icon(game_path + "icon", "wb+");
```
Comment (archshift) [inline src/core/hle/service/fs/archive.cpp]: `icon_file` might be a bit more descriptive?

**Comment by archshift** [comment]
LGTM

**Comment by bunnei** [comment]
LGTM

**Comment by yuriks** [comment]
LGTM

**Pull request merged**